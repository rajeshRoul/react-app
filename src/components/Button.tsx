import '../styles/Button.css'
import React, { ReactElement } from 'react'

type ButtonPropType = {
    handleClick: () => void
    value: string
}

function Button(props: ButtonPropType): ReactElement {
    const { handleClick, value } = props
    return (
        <button className="Button" type="button" onClick={handleClick}>
            {value}
        </button>
    )
}

export default Button
