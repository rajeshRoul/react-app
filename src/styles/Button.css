.Button {
    background-color: green;
    font-size: 1.5rem;
    color: white;
    border-radius: 3px;
    outline: none;
    border: 1px solid black;
    padding: 2px 5px;
}
