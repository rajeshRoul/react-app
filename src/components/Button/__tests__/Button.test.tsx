/* eslint-disable no-undef */
import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import Button from '../Button'
import '@testing-library/jest-dom/extend-expect'

afterEach(cleanup)

test('button click', () => {
    const handleClick = jest.fn()
    const { getByTestId } = render(
        <Button value="test button" handleClick={handleClick} />
    )
    expect(getByTestId('button')).toHaveTextContent('test button')
    fireEvent.click(getByTestId('button'))
    expect(handleClick).toHaveBeenCalledTimes(1)
    fireEvent.click(getByTestId('button'))
    expect(handleClick).toHaveBeenCalledTimes(2)
})

it('should take a snapshot', () => {
    const { getByTestId } = render(
        <Button value="test button" handleClick={() => {}} />
    )
    expect(getByTestId('button')).toMatchSnapshot()
})
