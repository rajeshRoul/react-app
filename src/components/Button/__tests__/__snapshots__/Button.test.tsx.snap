// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<button
  class="Button"
  data-testid="button"
  type="button"
>
  test button
</button>
`;
